lms/djangoapps/learner_dashboard/api/v0/views.py

```diff
@@ -335,3 +342,45 @@ def get(self, request, program_uuid):
                 'credit_pathways': credit_pathways,
             }
         )
+
+
+class CourseRecommendationApiView(APIView):
+
+    """
+        **Example Request**
+
+        GET api/dashboard/v0/recommendation/courses/
+    """
+
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """
+            Retrieves course recommendations details of a user in a specified course.
+        """
+        user_id = request.user.id
+        is_control, course_keys = get_personalized_course_recommendations(user_id)
```

Comment:
the return value of `None` for `is_control` where API response is non 200 will break the logic below where we are emitting the segment event and returning the response [here](https://github.com/openedx/edx-platform/pull/30675/files#diff-a7ecb99a77f0ce821624c3f438862a75e6a4a9e3aef3e8618c003363049c390eR386)